release/models/acl/openconfig-packet-match.yang

```diff
@@ -432,4 +454,55 @@ module openconfig-packet-match {
     }
   }
 
+  grouping ip-icmp-header-fields-top {
+    description
+      "Top grouping for the icpm filtering";
+
+    container icmp {
+      description
+        "Top container for the icpm filtering";
+
+      container config {
+        description
+          "Configuration attributes for the icmp filtering";
+
+        uses ip-icmp-header-fields-config;
+      }
+
+      container state {
+        description
+          "State attributes for the icmp filtering";
+        config false;
+
+        uses ip-icmp-header-fields-config;
+      }
+    }
+  }
+
+  grouping ip-icmp-header-fields-config {
+    description
+      "Collection of ICMP header fields that can be
+       used to set up a match filter.";
+    leaf icmp-type {
+      type uint8;
+      description
+        "Also known as control messages.";
+      reference
+        "RFC 792: Internet Control Message Protocol
+         RFC 4443: Internet Control Message Protocol (ICMPv6)
+                   for Internet Protocol Version 6 (IPv6)
+                   Specification.";
+    }
+    leaf icmp-code {
+      type uint8;
```

Comment:
Should we have this as an `enumeration`? We have an internal model that has the definitions of the ICMPv4 and ICMPv6 types that I can propose as an addition if this makes sense.